config/config_test.go

```diff
@@ -743,6 +776,184 @@ func TestMigrateConfigPurposeOneTreatment(t *testing.T) {
 	}
 }
 
+func TestMigrateConfigTCF2PurposeEnabledFlags(t *testing.T) {
+	oldPurposeConfig := []byte(`
+      gdpr:
+        tcf2:
+          purpose1:
+            enabled: false
+          purpose2:
+            enabled: true
+          purpose3:
+            enabled: false
+          purpose4:
+            enabled: true
+          purpose5:
+            enabled: false
+          purpose6:
+            enabled: true
+          purpose7:
+            enabled: false
+          purpose8:
+            enabled: true
+          purpose9:
+            enabled: false
+          purpose10:
+            enabled: true
+    `)
+	newPurposeConfig := []byte(`
+      gdpr:
+        tcf2:
+          purpose1:
+            enforce_purpose: "full"
+          purpose2:
+            enforce_purpose: "no"
+          purpose3:
+            enforce_purpose: "full"
+          purpose4:
+            enforce_purpose: "no"
+          purpose5:
+            enforce_purpose: "full"
+          purpose6:
+            enforce_purpose: "no"
+          purpose7:
+            enforce_purpose: "full"
+          purpose8:
+            enforce_purpose: "no"
+          purpose9:
+            enforce_purpose: "full"
+          purpose10:
+            enforce_purpose: "no"
+    `)
+	mixedPurposeConfig := []byte(`
+      gdpr:
+        tcf2:
+          purpose1:
+            enabled: false
+            enforce_purpose: "full"
+          purpose2:
+            enabled: false
+            enforce_purpose: "full"
+          purpose3:
+            enabled: false
+            enforce_purpose: "full"
+          purpose4:
+            enabled: false
+            enforce_purpose: "full"
+          purpose5:
+            enabled: false
+            enforce_purpose: "full"
+          purpose6:
+            enabled: false
+            enforce_purpose: "full"
+          purpose7:
+            enabled: false
+            enforce_purpose: "full"
+          purpose8:
+            enabled: false
+            enforce_purpose: "full"
+          purpose9:
+            enabled: false
+            enforce_purpose: "full"
+          purpose10:
+            enabled: false
+            enforce_purpose: "full"
+    `)
+	tests := []struct {
```

Comment:
I'm not sure if the tests will be easier to read if the yaml is added to each test case. These variables are only used once each, so this looks like an organizational choice.